app/views/layouts/default.html.erb

```diff
@@ -28,7 +28,7 @@
       <% end %>
     <% end %>
     <% if @has_json %>
-      <link rel="alternate" type="application/json" title="JSON version of this page" href="<%=h url_for(request.params.merge(:format => 'json')) %>">
+      <link rel="alternate" type="application/json" title="JSON version of this page" href="<%= current_path_as_json %>">
```

Comment:
I had a quick go, but I couldn't replicate https://github.com/mysociety/alaveteli/issues/5214 for this.

How did you end up with the same error?

I think the commit message here is dependent on the context in 2cbd17a53393613f3ee4d4b0f8c00533c3c8bb4b, which might not be obvious in the future, so it would be worth just tweaking this so that it explains what is meant by "It was _also_ possible for non-sanitized parameters to be used here"

Reply:
Ah yes, this could do with the commit message being expanded. Basically using `request.params` by-passes the ActionController::Parameters sanitization checking. Without the `request.` we would see the same error as the admin/timeline page